docs/internals/optimizer.rst

```diff
@@ -20,17 +22,21 @@ the function calls. Similarly, if a function is
 side-effect free and its result is multiplied by zero, you can remove the function
 call completely.
 
+The codegen-based optimizer currently consists of a single step that makes the compiler use unchecked
+arithmetic when generating code for the counter variable increment of certain ``for`` loops.
```

Comment:
Actually, this is not the only step performed in the codegen. I've seen comments that indicate that `settings.optimizer.orderLiterals` is also a codegen-level optimization (in the legacy codegen):

https://github.com/ethereum/solidity/blob/7df949edb22598e9d946660020c5390d0ce81bf8/libsolidity/interface/OptimiserSettings.h#L134-L136

It's value is passed to `ExpressionCompiler`, not to the optimizer, which confirms that.

Reply:
I was not aware of `orderLiterals`.
In that case, I think we should also add (in a future PR) a doc section for that step under the `Codegen-Based Optimizer Module` section.

Reply:
Still, in this PR we should not say that the new loop step is the only codegen-level optimization. Here we could just mention that the other step exists without going into detail about it.

Reply:
Yes. I have rewritten the relevant sections to reflect that.